sources/api/thar-be-settings/src/service.rs

```diff
@@ -62,23 +85,21 @@ where
     Ok(setting_to_services_map)
 }
 
-/// Given a map of Setting to affected Service name, return a
-/// HashSet of affected service names
-fn get_affected_service_names(
-    setting_to_service_map: HashMap<String, Vec<String>>,
-) -> HashSet<String> {
-    // Build a HashSet of names of affected services
-    debug!("Building set of affected services");
-    let mut service_set: HashSet<String> = HashSet::new();
-    for (_, service_list) in setting_to_service_map {
+/// Converts map of changed setting to affected service into affected service to changed setting
+fn get_service_setting_map(
+    setting_to_service_map: &HashMap<String, Vec<String>>,
+) -> HashMap<String, Vec<String>> {
+    let mut service_settings_map: HashMap<String, Vec<String>> = HashMap::new();
+
+    for (settings_name, service_list) in setting_to_service_map {
         for service_name in service_list {
-            debug!("Found {}", &service_name);
-            service_set.insert(service_name);
+            let settings = service_settings_map
+                .entry(service_name.clone())
+                .or_insert(vec![]);
```

Comment:
```suggestion
                .or_insert_with(Vec::new);
```
This way it won't create the Vec unless it's needed for a new entry.